### DISCUSSION TITLE
Unable to append to a list of tags using pynetbox

### DISCUSSION THREAD

**Discussion by pbolcar** [netbox-community/netbox#13045] [Q&A] [answered]
Hello, hopefully this is an appropriate place for this question.

I am trying to use pynetbox to add new tags to netbox virtualmachines/devices and I seem to be unable to add a new tag that is in the same format as the existing tags (the list items in no quotes vs my new item in quotes):

```
>>> device = nb.virtualization.virtual_machines.get(name='some_virtual_machine')

>>> device.tags
[tag_one, sometag, anothertag]

>>> device.tags.append(test)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'test' is not defined

>>> device.tags.append('test')

>>> device.tags
[tag_one, sometag, anothertag, 'test']

>>> device.save()
....pynetbox.core.query.RequestError: The request failed with code 400 Bad Request: {'tags': [{}, {}, {}, {}, {}, {}, {}, {}, {}, ['Related objects must be referenced by numeric ID or by dictionary of attributes. Received an unrecognized value: test']]}
```
It isn't clear to me how to add a tag without he quotes, or how netbox is currently storing tags. Perhaps I am missing something here? Thanks for any help or hints.

**Comment by markkuleinio** [comment, marked as answer]
First you need to create the tag:

```
nb.extras.tags.create({"name": "test", "slug": "test"})
```

Then you should be able to use:

```
device.tags.append({"name": "test"})
device.save()
```

https://majornetwork.net/2021/01/using-tags-with-pynetbox-in-netbox-2-9/

**Reply by pbolcar** [reply]
much thanks
